src/main/java/com/rybka/view/ExchangeView.java

```diff
@@ -8,33 +10,35 @@
 
     private final Scanner scanner = new Scanner(System.in);
     private final ExchangeService service = new ExchangeService();
+    private final HibernateDAO hibernateDAO = new HibernateDAO();
 
     public void showDialog() {
 
         try {
             System.out.print("Insert base currency: ");
-            String userBaseCurrencyAbbreviation = scanner.next().toUpperCase();
+            var userBaseCurrency = scanner.next().toUpperCase();
             System.out.print("Enter value: ");
-            Double userValue = scanner.nextDouble();
+            var userValue = scanner.nextDouble();
             System.out.print("Insert target currency: ");
 
-            String userTargetCurrencyAbbreviation = scanner.next().toUpperCase();
+            var userTargetCurrency = scanner.next().toUpperCase();
 
-            service.loadCurrencyOf(userBaseCurrencyAbbreviation, userTargetCurrencyAbbreviation);
-            service.exchange(userValue);
-            service.getCurrencyObject();
+            var currencyObject = service.loadCurrencyOf(userBaseCurrency);
+            System.out.println(currencyObject);
+            var currency = service.calculateTotal(currencyObject, userTargetCurrency, userValue);
```

Comment:
`calculateTotal` should only keep a calculation logic. for now it's responsible for calculation and creating a new object.

I'd do in such way:

var currencyResponse = service.loadCurrencyOf(userBaseCurrency);
var currency = retrieveTargetCurrency(currencyResponse, userTarget);
var total = service.calculateTotal(currency .getRate(), userValue); //return Double value
var convertedResult = constructConvertedResult(currencyObject.getBase/getTarget/getRate/total);

Reply:
fixed